### PR TITLE
Change Storage and Key Vault to track2

### PR DESCRIPTION
**Pull request by Luyunmt**
Change Storage and Key Vault to track2
https://github.com/Azure/Azure-Sentinel/issues/1063

### CODE REVIEW

**Comment by preetikr** [comment]
Hi @ehudk-msft - Can you and Alaa please review and close on this? This is a very old PR and we are trying to close these out. Thanks.

**Comment by jongio** [comment]
@preetikr  - Do you know who can help move this along?  Thanks

**Comment by v-xuto** [comment]
@scottaddie, @AlexGhiondea, @AlexGhiondea, @heaths - Could you help review this PR?

File: DataConnectors/O365 DataCSharp/Teams.CustomConnector.Common/KeyVaultHelper.cs
Comment (heaths) [inline DataConnectors/O365 DataCSharp/Teams.CustomConnector.Common/KeyVaultHelper.cs:70]: Nit: the null check is unnecessary - an exception would've been thrown - but it doesn't hurt either.

File: DataConnectors/O365 DataCSharp/Teams.CustomConnector.Common/Teams.CustomConnector.Common.csproj
Comment (heaths) [inline DataConnectors/O365 DataCSharp/Teams.CustomConnector.Common/Teams.CustomConnector.Common.csproj:8]: I highly recommend 1.4.1. There's been a lot of fixes and improvements since 1.2.3.

File: DataConnectors/O365 DataCSharp/Teams.CustomConnector.Common/Teams.CustomConnector.Common.csproj
Comment (heaths) [inline DataConnectors/O365 DataCSharp/Teams.CustomConnector.Common/Teams.CustomConnector.Common.csproj:9]: Nit: 4.2.0 is our latest, but is mostly just an update to the latest service version. This would still be supported. Just something to think about.

File: DataConnectors/O365 DataCSharp/Teams.CustomConnector.StorageHandler/StorageHandler.cs
Comment (heaths) [inline DataConnectors/O365 DataCSharp/Teams.CustomConnector.StorageHandler/StorageHandler.cs:104]: `StreamReader` is `IDisposable`
```suggestion
                        using StreamReader reader = new StreamReader(result.Value.Content);
```

**Review by heaths — APPROVED** [review]

File: DataConnectors/O365 DataCSharp/Teams.CustomConnector.Common/KeyVaultHelper.cs
Comment (Tianyun-G) [reply DataConnectors/O365 DataCSharp/Teams.CustomConnector.Common/KeyVaultHelper.cs]: @heaths - I have updated, please review.

File: DataConnectors/O365 DataCSharp/Teams.CustomConnector.Common/Teams.CustomConnector.Common.csproj
Comment (Tianyun-G) [reply DataConnectors/O365 DataCSharp/Teams.CustomConnector.Common/Teams.CustomConnector.Common.csproj]: @heaths - I have updated, please review

File: DataConnectors/O365 DataCSharp/Teams.CustomConnector.Common/Teams.CustomConnector.Common.csproj
Comment (Tianyun-G) [reply DataConnectors/O365 DataCSharp/Teams.CustomConnector.Common/Teams.CustomConnector.Common.csproj]: @heaths - I have updated, please review

**Review by heaths — APPROVED** [review]

**Comment by v-xuto** [comment]
@scottaddie, @AlexGhiondea, @AlexGhiondea- Could you help review this PR?

**Comment by heaths** [comment]
As the owner of the Azure Key Vault SDK for .NET, I signed off. The Storage SDK changes are also fine. We're all on the same team and work together. You probably don't need to wait for their reviews as well. 😉

**Comment by v-xuto** [comment]
@heaths  Could you help merge this PR?

**Comment by heaths** [comment]
I'm not a maintainer on this repo. I have no permissions to do so.

**Comment by jongio** [comment]
@shainw - Can you please review/merge?  Thanks!

**Comment by v-xuto** [comment]
@shainw - Can you please review/merge? Thanks!

**Pull request merged**
